pyfive/dataobjects.py

```diff
@@ -347,33 +347,61 @@ def filter_pipeline(self):
         offset = filter_msgs[0]['offset_to_message']
         version, nfilters = struct.unpack_from('<BB', self.msg_data, offset)
         offset += struct.calcsize('<BB')
-        if version != 1:
-            raise NotImplementedError("only version 1 filters supported. ")
-
-        res0, res1 = struct.unpack_from('<HI', self.msg_data, offset)
-        offset += struct.calcsize('<HI')
 
         filters = []
-        for _ in range(nfilters):
-            filter_info = _unpack_struct_from(
-                FILTER_PIPELINE_DESCR_V1, self.msg_data, offset)
-            offset += FILTER_PIPELINE_DESCR_V1_SIZE
-
-            padded_name_length = _padded_size(filter_info['name_length'], 8)
-            fmt = '<' + str(padded_name_length) + 's'
-            filter_name = struct.unpack_from(fmt, self.msg_data, offset)[0]
-            filter_info['filter_name'] = filter_name
-            offset += padded_name_length
-
-            fmt = '<' + str(filter_info['client_data_values']) + 'I'
-            client_data = struct.unpack_from(fmt, self.msg_data, offset)
-            filter_info['client_data'] = client_data
-            offset += 4 * filter_info['client_data_values']
-
-            if filter_info['client_data_values'] % 2:
-                offset += 4  # odd number of client data values padded
-
-            filters.append(filter_info)
+        if version == 1:
+            res0, res1 = struct.unpack_from('<HI', self.msg_data, offset)
```

Comment:
Just a side note, to not forget:

Would it be more convenient, if we move unpacking/offset-increment into a function?

We might also think about using a given structure (eg dict), which has the bits and pieces (name, unpacking information etc), to iterate over the `msg_data` and automatically unpack.

Reply:
Also wondering what `res0`, `res1` is in this context (probably need to immerse myself into HDF5 format). It is not needed anywhere else in this function.

Reply:
```res0``` and ```res1``` are nothing exciting... they are reserved sections and expected to contain zeros: https://docs.hdfgroup.org/hdf5/develop/_f_m_t3.html#FilterMessage

Reply:
I would agree that streamlining unpacking and moving the offset would be a useful thing to do - it is largely manual and error-prone (what if you forget to add the offset advance!).  There are a few places where you read but don't advance, but those could still be handled manually.